"Could not deserialize bitmap for key #0: error in roaringArray.readFrom: did not find expected serialCookie in header" on v1.8.0 when reading a bitmap written by roaring64

Per the [io.Reader](https://pkg.go.dev/io#Reader) documentation, it is acceptable for an implementation to read `0 <= n <= len(p)` bytes on each call to `Read`. I've found that `roaring64.Bitmap.ReadFrom` returns the following error when passed an `io.Reader` that reads less than `len(p)` bytes on calls to `Read`:
```
Could not deserialize bitmap for key #0: error in roaringArray.readFrom: did not find expected serialCookie in header
```

To replicate:
1. Create `main.go` with the following contents:
```
package main

import (
	"bytes"
	"io"

	"github.com/RoaringBitmap/roaring/roaring64"
)

func main() {
	bitmap := roaring64.NewBitmap()
	bitmap.Add(1234)

	var bitmapBuf bytes.Buffer
	if _, err := bitmap.WriteTo(&bitmapBuf); err != nil {
		panic(err)
	}

	bitmapMinusOneReader := minusOneReader{
		reader: &bitmapBuf,
	}
	readBitmap := roaring64.NewBitmap()
	if _, err := readBitmap.ReadFrom(&bitmapMinusOneReader); err != nil {
		panic(err)
	}
}

type minusOneReader struct {
	reader io.Reader
}

func (r minusOneReader) Read(p []byte) (int, error) {
	if len(p) > 1 {
		p = p[:len(p)-1]
	}
	return r.reader.Read(p)
}
```
2. Create `go.mod` with the following contents:
```
module roaring-replication

go 1.20

require github.com/RoaringBitmap/roaring v1.8.0

require (
	github.com/bits-and-blooms/bitset v1.12.0 // indirect
	github.com/mschoch/smat v0.2.0 // indirect
)
```
3. Run `go mod tidy`
4. Run `go run main.go`
5. Observe error from `ReadFrom`:
```
panic: Could not deserialize bitmap for key #0: error in roaringArray.readFrom: did not find expected serialCookie in header

goroutine 1 [running]:
main.main()
	main.go:24 +0x12d
exit status 2
```

--- a/roaring64/roaring64.go
+++ b/roaring64/roaring64.go
@@ -135,8 +135,8 @@ func (rb *Bitmap) FromUnsafeBytes(data []byte) (p int64, err error) {
 func (rb *Bitmap) ReadFrom(stream io.Reader) (p int64, err error) {
 	sizeBuf := make([]byte, 8)
 	var n int
-	n, err = stream.Read(sizeBuf)
-	if n == 0 || err != nil {
+	n, err = io.ReadFull(stream, sizeBuf)
+	if err != nil {
 		return int64(n), err
 	}
 	p += int64(n)
@@ -159,8 +159,8 @@ func (rb *Bitmap) ReadFrom(stream io.Reader) (p int64, err error) {
 	}
 	keyBuf := sizeBuf[:4]
 	for i := uint64(0); i < size; i++ {
-		n, err = stream.Read(keyBuf)
-		if n == 0 || err != nil {
+		n, err = io.ReadFull(stream, keyBuf)
+		if err != nil {
 			return int64(n), fmt.Errorf("error in bitmap.readFrom: could not read key #%d: %s", i, err)
 		}
 		p += int64(n)

